### PR TITLE
tooling: Add macro for generating templated content

### PR DESCRIPTION
**Pull request by phlax**
This macros provides an easy way to generate templated content from a jinja template and json/yaml configs

Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
This only provides the macro. I intend to follow up with PRs to add a dockerhub readme #25550, improve the release messaging on the gh release page and a few other uses

File: tools/base/envoy_python.bzl [hunk 1]
```diff
@@ -333,3 +333,100 @@ def envoy_py_data(name, src, format = None, entry_point = base_entry_point):
...
+        env_kwargs = template_kwargs,
+        templates = [template],
+        entry_point = entry_point,
+    )
+    native.genrule(
+        name = "%s_generate_content_py" % name,
+        cmd = """
+        echo "import pathlib" > $@ \
+        && echo "import sys" >> $@ \
+        && echo "import os" >> $@ \
```
Comment (phlax) [inline tools/base/envoy_python.bzl:403]: unused

File: tools/base/envoy_python.bzl [hunk 2]
```diff
@@ -333,3 +333,100 @@ def envoy_py_data(name, src, format = None, entry_point = base_entry_point):
...
+        && echo "DATA_LIB = \\"$(location :%s)\\"" >> $@ \
+        && echo "sys.path.append(pathlib.Path(DATA_LIB).parent)" >> $@ \
+        && echo "from %s import data" >> $@ \
+        && echo "TPL_LIB = \\"$(location :%s)\\"" >> $@ \
+        && echo "sys.path.append(pathlib.Path(TPL_LIB).parent)" >> $@ \
+        && echo "from %s import env" >> $@ \
+        && echo "print(env.get_template(\\"%s\\").render(**data))" >> $@
+        """ % (name_data, name_data, name_tpl, name_tpl, template_name),
+        outs = ["%s_generate_content.py" % name],
+        exec_tools = [":%s" % name_data, name_tpl, template],
```
Comment (phlax) [inline tools/base/envoy_python.bzl:413]: probs use `tools`

File: tools/base/envoy_python.bzl [hunk 3]
```diff
@@ -333,3 +333,100 @@ def envoy_py_data(name, src, format = None, entry_point = base_entry_point):
...
+    )
+    py_binary(
+        name = "%s_generate_content" % name,
+        main = ":%s_generate_content.py" % name,
+        srcs = [":%s_generate_content.py" % name],
+        deps = [
+            ":%s" % name_data,
+            name_tpl,
+            requirement("envoy.base.utils"),
```
Comment (phlax) [inline tools/base/envoy_python.bzl:422]: doesnt seem to be required

File: tools/base/envoy_python.bzl [hunk 4]
```diff
@@ -333,3 +333,100 @@ def envoy_py_data(name, src, format = None, entry_point = base_entry_point):
...
+        name = "%s_generate_content_py" % name,
+        cmd = """
+        echo "import pathlib" > $@ \
+        && echo "import sys" >> $@ \
+        && echo "import os" >> $@ \
+        && echo "DATA_LIB = \\"$(location :%s)\\"" >> $@ \
+        && echo "sys.path.append(pathlib.Path(DATA_LIB).parent)" >> $@ \
+        && echo "from %s import data" >> $@ \
+        && echo "TPL_LIB = \\"$(location :%s)\\"" >> $@ \
+        && echo "sys.path.append(pathlib.Path(TPL_LIB).parent)" >> $@ \
```
Comment (phlax) [inline tools/base/envoy_python.bzl:408]: fold these lines - ditto above

File: tools/base/envoy_python.bzl [hunk 5]
```diff
@@ -333,3 +333,96 @@ def envoy_py_data(name, src, format = None, entry_point = base_entry_point):
...
+    )
+    envoy_py_data(
+        name = "%s_data" % name,
+        src = ":%s_json" % name,
+        entry_point = entry_point,
+    )
+    envoy_jinja_env(
+        name = name_tpl,
+        env_kwargs = template_kwargs,
+        templates = [template],
```
Comment (phlax) [inline tools/base/envoy_python.bzl:395]: add `template_filters`

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!

**Pull request merged**